pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -786,6 +801,13 @@ func (r *ReconcileLogStorage) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	dpiList := &v3.DeepPacketInspectionList{}
```

Comment:
You need to check on whether the DPI watch is ready before attempting to query it - if we don't wait, then we'll potentially hit the API server with lots of Get() requests.

The operator is designed to serve Get() requests from its cache, and only ever perform Watches against the API server.